.travis.yml

```diff
@@ -0,0 +1,5 @@
+language: java
+jdk: oraclejdk9
+install: ./gradlew --version
+script: ./gradlew --continue --init-script gradle/init-scripts/build-scan.init.gradle.kts --scan --stacktrace sanityCheck
+if: pull_request AND head_repo != gradle/gradle
```

Comment:
Shall we do some caching of the Gradle user home? Something like:
```
before_cache:
- rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
- rm -fr $HOME/.gradle/caches/*/plugin-resolution/
- rm -fr $HOME/.gradle/caches/*/scripts/
cache:
  directories:
  - $HOME/.gradle/caches/
  - $HOME/.gradle/wrapper/
```

We do that for example in the guides repositories
https://github.com/gradle-guides/using-build-cache/blob/057d485b930dd4946ec2c0f6841a59945cba6594/.travis.yml#L5-L12
```
before_cache:
- rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
- rm -fr $HOME/.gradle/caches/*/plugin-resolution/
- rm -fr $HOME/.gradle/caches/*/scripts/
cache:
  directories:
  - $HOME/.gradle/caches/
  - $HOME/.gradle/wrapper/
```

Reply:
Travis' cache is really limited. For starters it's not incremental, i.e. a single changed/added/removed file will cause it to create a new archive of all cached directories and upload it to its cache server.

The distribution download takes 4 seconds and since we change that more often than a typical project with a Gradle build, I wouldn't cache it. Caching dependencies might make sense but zipping 419 MB (according to the build scan) and extracting it in the beginning might also do more harm than good. Downloading dependencies takes 3 of the 28 total minutes of build time. So, even if caching came at no cost, we wouldn't save much. Thus, I'd omit caching for now.